### PR TITLE
planet size calculation #391 fixes

### PR DESCRIPTION
**Pull request by yazilimmz**
I made changes based on [this page](https://ogame.fandom.com/wiki/Colonization). Adjustments were made according to the first planet.

### CODE REVIEW

File: app/Factories/PlanetServiceFactory.php [hunk 1]
```diff
@@ -428,13 +430,17 @@ private function setupMoonProperties(Planet $planet): void
...
-        $planet->diameter = 300;
-        $planet->field_max = rand(140, 250);
+
+        $planet_data = $this->planetData($planet->planet, $is_first_planet);
+
+        // Random field count between the min and max values and add the Server planet fields bonus setting.
+        $planet->field_max = rand($planet_data['fields'][0], $planet_data['fields'][1]) + $this->settings->planetFieldsBonus();
+        $planet->diameter = (int) (36.14 * $planet->field_max + 5697.23);
 
         // TODO: temperature range should be dependent on the planet position.
```
Comment (lanedirt) [inline app/Factories/PlanetServiceFactory.php]: Can you remove this todo line since its now implemented

File: app/Factories/PlanetServiceFactory.php [hunk 2]
```diff
@@ -443,11 +449,232 @@ private function setupPlanetProperties(Planet $planet): void
...
-        $planet->deuterium_synthesizer_percent = 10;
-        $planet->solar_plant_percent = 10;
-        $planet->fusion_plant_percent = 10;
-        $planet->solar_satellite_percent = 10;
+        $planet->metal_mine_percent = $planet_data['metal_mine_percent'];
+        $planet->crystal_mine_percent = $planet_data['crystal_mine_percent'];
+        $planet->deuterium_synthesizer_percent = $planet_data['deuterium_synthesizer_percent'];
+        $planet->solar_plant_percent = $planet_data['solar_plant_percent'];
+        $planet->fusion_plant_percent = $planet_data['fusion_plant_percent'];
+        $planet->solar_satellite_percent = $planet_data['solar_satellite_percent'];
```
Comment (lanedirt) [inline app/Factories/PlanetServiceFactory.php]: Currently in OGameX the `*_percent` fields in the `planets` table are associated with the production percentage that the player can configure on the resource settings page, example: 
Screenshot 2024-12-23 at 21 30 47

If I read your code correctly then based on the position of the planet a bonus modifier should be added to the mine output. However to make this work we'll probably need to add this bonus modifier to the resource production formula. See `PlanetService::updateResourceProductionStatsInner()` for how this is currently implemented.

So for this PR I would propose to:
1. Revert this specific change back to the original version (set everything to "10" which represents 100% in the screenshot above)
2. Integrate the planet resource bonus modifier into the `PlanetService::updateResourceProductionStatsInner()` logic. However as this is outside of the scope of #391 we could also make a new issue for this to take on separately.

**Review by lanedirt — CHANGES_REQUESTED** [review]
Hi @yazilimmz,

Thanks for your contribution and for looking into implementing the planet size calculations! 

Overall the changes to the field calculation looks good to me. I reviewed the changes and did add two comments, see below. Could you take a look at these?

If you have any questions feel free to let me know. Thanks again for your time in helping out, much appreciated! 🚀 😃

**Comment by yazilimmz** [comment]
Hello,
Thank you for starting and leading this project; we truly appreciate your efforts.
You're absolutely right—I hadn't realized that the percentage pointed there. I've made adjustments accordingly.

Using this link:
https://proxyforgame.com/en/ogame/calc/production.php

I noticed that it corresponds to the base production, so I updated it based on that.
If you could create an issue and assign it to me for calculating energy and deuterium based on the planet's temperature, I’d be happy to look into it as well.

File: app/Services/PlanetService.php [hunk 3]
```diff
@@ -1462,6 +1520,7 @@ public function getPlanetBasicIncome(): Resources
      */
     private function updateResourceProductionStatsInner(Resources $production_total, int|float $energy_production_total, int|float $energy_consumption_total, bool $save_planet = true): void
     {
+
```
Comment (lanedirt) [inline app/Services/PlanetService.php]: It looks like nothing changed in this method `updateResourceProductionStatsInner()` except adding and removing a newline, can you revert the change on line 1523 and line 1482?

Comment (yazilimmz) [reply app/Services/PlanetService.php]: > It looks like nothing changed in this method `updateResourceProductionStatsInner()` except adding and removing a newline, can you revert the change on line 1523 and line 1482?

I have updated the code again for your reference. Since the production calculations are added to the base production, you can see it in PlanetService::getPlanetBasicIncome().

**Comment by lanedirt** [comment]
Hi @yazilimmz,

Thanks for the adjustments, functionally it looks good to me now. I noticed that some tests were failing, but as I was typing this I saw your new commit come in and now it looks like its working properly, nice 😄! Will merge this PR now. Welcome as an official contributor to the OGameX project 🚀 !

For the other changes you mentioned earlier regarding energy and deuterium output an issue already exists, see:
- #310

If you have the time to look at this it would be greatly appreciated :-)
